src/channel.ts

```diff
@@ -1530,47 +1521,47 @@ export class Channel<
       },
     );
 
-    const s = channel.state;
+    const channelState = channel.state;
     switch (event.type) {
       case 'typing.start':
         if (event.user?.id) {
-          s.typing = s.typing.set(event.user.id, Immutable(event));
+          channelState.typing[event.user.id] = event;
         }
         break;
       case 'typing.stop':
         if (event.user?.id) {
-          s.typing = s.typing.without(event.user.id);
+          delete channelState.typing[event.user.id];
         }
         break;
       case 'message.read':
         if (event.user?.id) {
-          s.read = s.read.set(
-            event.user.id,
-            Immutable({ user: { ...event.user }, last_read: event.received_at }),
-          );
+          channelState.read[event.user.id] = {
+            last_read: event.received_at ? new Date(event.received_at) : new Date(),
```

Comment:
the channel state read object expects `last_read` to be `Date` type and the `event.received_at` type is `Date | string | undefined`. so we can't keep it the same. when the Immutable was being used this was not showing any TS issue because the object being used had `any`/`unknown` as the type